components/script/dom/console.rs

```diff
@@ -27,74 +28,96 @@ impl Console {
     }
 }
 
+fn with_stderr_lock<F>(f: F) where F: FnOnce() {
+    let stderr = io::stderr();
+    let _handle = stderr.lock();
+    f()
+}
+
 impl Console {
     // https://developer.mozilla.org/en-US/docs/Web/API/Console/log
     pub fn Log(global: &GlobalScope, messages: Vec<DOMString>) {
-        for message in messages {
-            println!("{}", message);
-            Self::send_to_devtools(global, LogLevel::Log, message);
-        }
+        with_stderr_lock(move || {
+            for message in messages {
```

Comment:
Also, I wonder whether we should split this loop (and the others) into two, one that does the printing with the lock held, and other that calls `send_to_devtools`.

I say that, because as far as I can tell, that function can panic and log (it uses `downcast`, which calls `get_dom_class`, which does `trace!`, which writes to stderr I think). I'd expect this to deadlock with `RUST_LOG=trace`, wouldn't it? or is the lock re-entrant? If it is, please comment on it.